core/templates/pages/preferences-page/preferences-page.component.spec.ts

```diff
@@ -31,63 +26,202 @@ import { TranslatorProviderForTests } from 'tests/test.extras';
     ctrl.$onInit();
+    $rootScope.$apply();
+    $httpBackend.flush();
   }));
 
-  it('should show that editor role notifications checkbox is true by default',
-    function() {
-      $httpBackend.flush();
-      expect(ctrl.canReceiveEditorRoleEmail).toBe(true);
-    });
+  it('should get static image url', function() {
+    expect(ctrl.getStaticImageUrl('/path/to/image.png')).toBe(
+      '/assets/images/path/to/image.png');
+  });
 
-  it('should show that feedback message notifications checkbox is true' +
-    'by default',
-  function() {
+  it('should save user bio', function() {
+    var userBio = 'User bio example';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match('"update_type":"user_bio"');
+    };
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.saveUserBio(userBio);
+    $httpBackend.flush();
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should save subject interests after changing it', function() {
+    var subjectInterests = 'Math';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"subject_interests"');
+    };
+
+    expect(ctrl.subjectInterestsChangedAtLeastOnce).toBe(false);
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.onSubjectInterestsSelectionChange(subjectInterests);
+    $httpBackend.flush();
+
+    expect(ctrl.subjectInterestsChangedAtLeastOnce).toBe(true);
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should save preferred site language code after changing it', function() {
+    var newLanguage = 'es';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"preferred_site_language_code"');
+    };
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.savePreferredSiteLanguageCodes(newLanguage);
+    $httpBackend.flush();
+
+    expect(ctrl.select2DropdownIsShown).toBe(false);
+    $timeout.flush();
+    expect(ctrl.select2DropdownIsShown).toBe(true);
+
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('should save preferred audio language code after changing it', function() {
+    var newLanguage = 'es';
+    var isRequestTheExpectOne = function(queryParams) {
+      return decodeURIComponent(queryParams).match(
+        '"update_type":"preferred_audio_language_code"');
+    };
+
+    $httpBackend.expect(
+      'PUT', '/preferenceshandler/data', isRequestTheExpectOne).respond(200);
+    ctrl.savePreferredAudioLanguageCode(newLanguage);
```

Comment:
Check that this was set correctly. Currently there is no assertion after this call...

Reply:
This call doesn't change any variable or property of the controllers. It only makes an HTTP request with no promise handling, so actually I think there is nothing to really assert. The "assertion" I did is to check if there is an HTTP request to be handle. Here and elsewhere which follows this case.

Reply:
Oh ok.. Sounds good...